src/LanguageServer.php

```diff
@@ -206,6 +206,13 @@ public function initialize(ClientCapabilities $capabilities, string $rootPath =
                 // Find composer.json
                 if ($this->composerJson === null) {
                     $composerJsonFiles = yield $this->filesFinder->find(Path::makeAbsolute('**/composer.json', $rootPath));
+
+                    // If we sort our findings by string length (shortest to longest),
+                    // the first entry will be the project's root composer.json.
```

Comment:
Yes, that would work! 👍 
But you need to apply it to the `path` component of the URI, otherwise a URL with host will have 2 more slashes than a URN. Use `Uri\parse()`